autoscaling.tf

```diff
@@ -0,0 +1,111 @@
+resource "aws_appautoscaling_target" "table_read" {
+  count              = var.autoscaling_read_max_capacity > 0 ? 1 : 0
+  max_capacity       = var.autoscaling_read_max_capacity
+  min_capacity       = var.read_capacity
+  resource_id        = "table/${var.name}"
+  scalable_dimension = "dynamodb:table:ReadCapacityUnits"
+  service_namespace  = "dynamodb"
+}
+
+resource "aws_appautoscaling_policy" "table_read_policy" {
+  count              = var.autoscaling_read_max_capacity > 0 ? 1 : 0
+  name               = "DynamoDBReadCapacityUtilization:${aws_appautoscaling_target.table_read[0].resource_id}"
+  policy_type        = "TargetTrackingScaling"
+  resource_id        = aws_appautoscaling_target.table_read[0].resource_id
+  scalable_dimension = aws_appautoscaling_target.table_read[0].scalable_dimension
+  service_namespace  = aws_appautoscaling_target.table_read[0].service_namespace
+
+  target_tracking_scaling_policy_configuration {
+    predefined_metric_specification {
+      predefined_metric_type = "DynamoDBReadCapacityUtilization"
+    }
+
+    scale_in_cooldown  = var.autoscaling_scale_in_cooldown
+    scale_out_cooldown = var.autoscaling_scale_out_cooldown
+    target_value       = var.autoscaling_target_value
```

Comment:
Maybe there should be multiple variables for `autoscaling_target_value`? One for each type of policy?

Reply:
You mean one of read and one for write? It's conceivable, yes. OK I'll split it into 2 vars.

Reply:
I think splitting into 4 (one for each `aws_appautoscaling_policy` resource) and maybe use the master one `var.autoscaling_target_value` as the default one.

Reply:
Maybe I split it into 2 maps then because there's a few attributes for each of read and write configuration. e.g.

```hcl
autoscaling_write = {
  max_capacity = 20
  scale_in_cooldown = 60
  scale_out_cooldown = 120
  target_value = 75
}
```

?

Reply:
yes, this is even better than what I thought :)

Reply:
OK done